### PR TITLE
Update Money Lost Info Form

### LINKED ISSUE #2154: Update Money Lost Info Form
Update MoneyLostInfoForm.js to use Formik, Yup, and Bootstrap as documented in issue #2136. These changes have been made to HowDidItStartForm.js, you can them here - #2138.  

A new component will need to be created for the input element (if not already created). Documentation can be found [here](https://react-bootstrap.github.io/components/forms/#forms-controls).

This form uses date validation, this should be refactored to use Yup - [Yup Date](https://github.com/jquense/yup#date) and [Yup Extend Type](https://github.com/jquense/yup#extending-schema-types).

### PR DESCRIPTION
**Pull request by kevinyang06**
Fixes #2154

# Description

> Update MoneyLostInfoForm.js to use Formik, Yup, and Bootstrap as documented in issue #2136. These changes have been made to HowDidItStartForm.js, you can them here - #2138.

A new component will need to be created for the input element (if not already created). Documentation can be found here.

This form uses date validation, this should be refactored to use Yup - Yup Date and Yup Extend Type.

# Any new packages installed?

> Give details

# Required followup work

> Is there anything related to this that still needs to be done (ex: documentation changes).

# Checklist:

- [ ] I have updated the azurescript.sh with any **new environment variables** and added them to the appsettings
- [ ] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [ ] I have added and needed tests for my changes (in particular for new screens)
- [ ] I have added a comment to any confusing code
- [ ] I have added documentation to any modified front-end code. (Or added missing documentation)

### CODE REVIEW

**Comment by justinr86** [comment]
The space in between form questions is too big.
MoneyForm

**Comment by justinr86** [comment]
It looks like the date input is not inline with the rest of the form.

MoneyForm2

**Comment by justinr86** [comment]
It looks like you lost some French message properties.

MoneyForm3

**Comment by kevinyang06** [comment]
> It looks like you lost some French message properties.
> 
> MoneyForm3

fixed

**Comment by kevinyang06** [comment]
> The space in between form questions is too big.
> MoneyForm

Is it ok to leave the space between "how much money did suspect steal" and "how was money send to them"? otherwise the select method of payment will disconnect with options down below.
image
image

**Comment by ying-chen-cds** [comment]
This should be a text input, not a text area
moneyLost

**Comment by ying-chen-cds** [comment]
Need to have validation for the date. Right now, no validation at all.

**Comment by kevinyang06** [comment]
> This should be a text input, not a text area
> moneyLost

fixed

**Comment by kevinyang06** [comment]
> Need to have validation for the date. Right now, no validation at all.

fixed

**Comment by justinr86** [comment]
Looks like you have an additional label at the top of the page.

moneyLostLabel

**Comment by justinr86** [comment]
Validation allows empty year.

badDate

**Comment by ying-chen-cds** [comment]
Not sure where this question coming from? 
moneyLost

**Comment by ying-chen-cds** [comment]
Date validation is not working properly. If you just enter day without month, year, it will let you pass, that's not right.

**Comment by kevinyang06** [comment]
> Date validation is not working properly. If you just enter day without month, year, it will let you pass, that's not right.

fixed

**Comment by kevinyang06** [comment]
> Not sure where this question coming from?
> moneyLost

fixed

File: f2/src/forms/MoneyLostInfoForm.js [hunk 1]
```diff
@@ -2,69 +2,68 @@
...
-    'methodPayment.creditCard',
-    'methodPayment.giftCard',
-    'methodPayment.cryptocurrency',
-    'methodPayment.other',
+  const formOptions = [
+    {
+      name: 'eTransfer',
+      value: 'methodPayment.eTransfer',
+      checkboxLabel: <Trans id="methodPayment.eTransfer" />,
+      checkboxName: 'methodPayment.eTransfer',
```
Comment (justinr86) [inline f2/src/forms/MoneyLostInfoForm.js:35]: The checkboxName property is never used.

**Review by sdzhangtao — APPROVED** [review]

**Pull request merged**